### ISSUE TITLE
Flatpack Steam won't let me instal games on a secondary drive

### ISSUE THREAD

**Issue by BlackStark62** [flathub/com.valvesoftware.Steam#1183]
I am currently running Linux mint 21.1 vera and one of the recent updates to the flatpack version of steam has made it so that I cannot install games to my secondary drive where I intend to store them. Even after taking steps to make sure make sure the disk has appropriate permissions. When I make a new library on that drive and try to install a game it stops and reports a disk write error every time.

**Comment by tsoruu** [comment]
Same here, Want to make a unified Steam Library for Linux and Windows in an external drive, formated in NTFS. Running VanillaOS, which should be irrelevant (from my understanding) as Flatpak runs similarly to a container... Adding directory via
`flatpak override --user --filesystem=/path/to/mountpoint com.valvesoftware.Steam`
and adding the directory in Steam as an external drive didn't work as it shows that Steam binds the actively selected drive as /run/user/1000/doc/e639edc2/steamlibrary. Should I mount the drive first or should I just appoint the drive to /dev/nvme01px? Adding the mounted file system in the flatpak override seems like Steam just skips the Drive discovery as there isn't any error code nor the added drive in the Storage setting of Steam

**Comment by drguildo** [comment]
I also have this issue on Fedora 38. I have done `flatpak override --user --filesystem=/mnt/foo com.valvesoftware.Steam` and restarted Steam, but when I go to either add a new library, or create a new one, under /mnt/foo, it just silently fails.

If I create a new one, it seems to create the steamapps subdirectory, but it doesn't actually list it in the Steam GUI.

**Comment by Gamertech** [comment]
Mount the drive (/mnt/games for me), then add that path to Steam via Flatseal.

If flatpak apps aren't allowed to access something directly, paths show up as /run/user/etc.

**Comment by yanickrochon** [comment]
I have been trying to find a solution for this for a while, now.

The exact path to my external Steam library is set, and starting Steam I can indeed add the external library, however it can't see any game.

![image](https://github.com/flathub/com.valvesoftware.Steam/assets/78461/02abf803-0aa6-4b0a-8d70-d025242182d0)

All games look like they aren't installed. When I try to reinstall one, the process hangs when downloading data. It seems like Steam has no read access to the external library?

When I browse the files of that library, it correctly opens Nautilus in the `steamapps` folder, and I can see all the `appmanifest_???.acf` files.

**Comment by Gamertech** [comment]
> I have been trying to find a solution for this for a while, now.

You have to whitelist additional drives in the flatpak sandbox. The /run/user location is a temporary address punched through xdg-portals because you didn't allow access. It wont work.

- Ensure the drive is permanently mounted at /mnt/games or somewhere appropriate
- Download the Flatseal flatpak and exit Steam
- Run Flatseal, select Steam, scroll down to Filesystem, and in Other Files, add the exact /mnt/games or whatever mount
- Launch Steam and configure the storage drive

**Comment by yanickrochon** [comment]
@Gamertech thank you for the response, however this is what I did. But, now I have the other problem described, here : https://github.com/flatpak/xdg-desktop-portal/issues/689 where Steam won't show my files, but `xdg-docum`, somehow, has these files mounted and locked even is Steam is no longer running.

Well, this is not helping making Linux welcoming for Windows users!

**Comment by Oroborius** [comment]
@rairay91 Unfortunately this does not work for me. However I am using a different setup so maybe that's why?

Currently trying to get Steam to load a partition that belongs to an LVM while the partition is an XFS partition. Furthermore it's almost 4TB. Anyways I cannot for the life of me figure out how to get this to load into Steam.

**Comment by rairay91** [comment]
If you are talking about granting read and right permission to that directory/drive from flatseel then you this permission is required for any flatpak app(which you will do by appending r:o after i guess.

While if you are talking about granting read and write permission for directory/drive from /etc/fstab file then in the screenshot above i have shown my drive is mounted with couple of arguments/options, one of them is users, which tell system that user can read and right to this drive without any ristriction. You also need to add others(defaults,exec,auto) in the same order.
This let your application(steam from flatpak as well as system package) to access drive, directory, symlink in any way. You can add directory or entire partition or symlink to directory/drive, whatever way you want to go.
It took me days to find out the solution, and not a single solution online was complete, but finally this one worked out of the box.

**Comment by Korkman** [comment]
@rairay91 Not quite correct. The `users` mount option allows any user to _mount_ the filesystem, it does not grant write permissions - this is what `chmod` is for.

**Comment by Novantric** [comment]
All the above ain't workin for me...

**Comment by 12sunflowers** [comment]
Same here flatpak Steam client not detect games at ntfs disk with SteamLibrary
already did
`flatpak override --user --filesystem=/media/freeartist-devuan/NEW4/SteamLibrary com.valvesoftware.Steam`
not help
![Image](https://github.com/user-attachments/assets/a42aaf5f-e6dd-4eea-8c6f-835fce190e84)

**Comment by yanickrochon** [comment]
> Same here flatpak Steam client not detect games at ntfs disk with SteamLibrary already did `flatpak override --user --filesystem=/media/freeartist-devuan/NEW4/SteamLibrary com.valvesoftware.Steam` not help 

My issue was write permission on the disk. For some reason, the disk was mounted RO

**Comment by 12sunflowers** [comment]
@yanickrochon 
Thanks for quick response
I tried 2 methods mound:
1. via Thunar
2. via fstab 
`UUID=4E9E2ABD9E2A9E09 /media/freeartist-devuan/NEW4   ntfs        nofail,users,defaults,exec,auto     0       0`
both without RO, same not shown games(

**Comment by 12sunflowers** [comment]
Its very wired
redo again
```
freeartist-devuan@devuan:~$ flatpak override --user --filesystem=/media/freeartist-devuan/NEW4/SteamLibrary com.valvesoftware.Steam
freeartist-devuan@devuan:~$ flatpak override --user --filesystem=/media/freeartist-devuan/NEW2/Steam com.valvesoftware.Steam
```
1.mount drive via Thunar
2.launch Steam
3.at Steam storage use this path for add drive(see screenshot)

![Image](https://github.com/user-attachments/assets/2b05be62-cb50-43a9-9e02-eaec277f9900)
![Image](https://github.com/user-attachments/assets/6646c0ef-6457-46a1-ab20-f10fb02558f5)

**Comment by SuperSlayer21** [comment]
> > He estado tratando de encontrar una solución para esto por un tiempo, ahora.
> 
> Tienes que una lista blanca de unidades adicionales en la caja de arena Flatpak. La ubicación /ejecución /usuario es una dirección temporal perforada a través de los portales XDG porque no permitió el acceso. No funcionará.
> 
> * Asegúrese de que la unidad esté montada permanentemente en /mnt /juegos o en algún lugar apropiado
> * Descargue el plano plano plano y salga de vapor
> * Ejecute FlatSeal, seleccione Steam, desplácese hacia abajo al sistema de archivos y en otros archivos, agregue los juegos exactos /mnt /o cualquier monte
> * Iniciar vapor y configurar la unidad de almacenamiento

Muchas Gracias bro me funciono tu solucion Gracias

**Comment by xrishox** [comment]
I just thought I would point out something that was not obvious to me and would have saved me hours of headaches.

On a traditional linux install where steam is installed natively the way you add the drive is by adding for example /mnt/Steam which is the root of the drive. This did not work for me on the flatpak version. What I had to do in the flatpak version (after passing it through with flatseal) was point it to /mnt/Steam/SteamLibrary and then it worked immediately

**Comment by Novantric** [comment]
> What I had to do in the flatpak version (after passing it through with flatseal) was point it to /mnt/Steam/SteamLibrary and then it worked immediately

Does this work no matter what the drive is? If so it could be added as a default flatpak permission.

**Comment by trr** [comment]
> Does this work no matter what the drive is? If so it could be added as a default flatpak permission.

No, because this feature in Steam is for adding a non-standard directory containing a user's steam library, so by definition we can't predict where that will be.  /mnt/Steam/SteamLibrary is not a standardised filesystem location - it was just an example of a possible location where a user may have chosen to have their steam library.  Mine's in ~/Games/SteamLibrary

**Comment by Aasikki** [comment]
For anyone still having issues after having the correct mount options and passing it through flatseal, try repairing the library. Not sure if just a coincidence, but after that downloads started working for me.

<img width="715" height="391" alt="Image" src="https://github.com/user-attachments/assets/8d06ff97-d315-4551-99d2-e23c94d10ef0" />
